src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogSummaryMailGenerator.java

```diff
@@ -102,16 +105,32 @@ public String prepareMailContent(int numberOfDaysBackInThePast, long issueNumber
         ).collect(Collectors.toList());
     }
 
-    private Map<Blog, String> getBlogAndItsHomepage(List<Blog> blogsAddedSinceLastNewsletter) {
-        return blogsAddedSinceLastNewsletter.stream().collect(Collectors.toMap(
-            Function.identity(),
-            blog -> getBlogHomepageFromRss(blog.getRss()))
-        );
+    private Map<Blog, HomePageUrl> getBlogAndItsHomePage(
+        List<Blog> blogsAddedSinceLastNewsletter) {
+        return blogsAddedSinceLastNewsletter.stream()
+            .map(blog -> Pair.of(blog,
```

Comment:
Yeah, the first line is too long so formatter handled this case as best as he can and I screwed up the second part, but it will be fixed soon. However, I can do nothing about the function signature formatting - it is simply too long.